server/src/controllers/middleware/getFavorite.js

```diff
@@ -0,0 +1,28 @@
+const { getFavoriteCourse, getUserById } = require('../../database/queries');
+
+const getFavorite = async (req, res, next) => {
+  try {
+    const { userId } = req.params;
+    if (userId > 0) {
+      const checkUser = await getUserById(userId);
+      if (checkUser.rowCount) {
+        const { rows } = await getFavoriteCourse(userId);
+        if (rows[0]) {
+          res.json(rows);
+        } else {
+          res.json({
+            message: 'There is no favorite courses for this user id',
+          });
+        }
+      } else {
+        res.status(404).json({
+          message: 'Invalid id',
+        });
+      }
+    }
```

Comment:
if `userId` is less than or equal to zero no response will return
remove the check `if (userId > 0) { ... } `
OR add `else` case
```js
if (userId > 0) {
    ...
} else {
    res.status(400).json({
        message: 'invalid user id',
    });
}
```

Reply:
yes, thank you I forget to make a schema for this params.